### ISSUE TITLE
r8169 module not working

### ISSUE THREAD

**Issue by getchoo** [pop-os/pop#1808]
**Distribution (run `cat /etc/os-release`):** Pop!_OS 21.04
 
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
```
linux-modules-5.11.0-7620-generic:
  Installed: 5.11.0-7620.21~1624379747~21.04~3abeff8
  Candidate: 5.11.0-7620.21~1624379747~21.04~3abeff8
  Version table:
 *** 5.11.0-7620.21~1624379747~21.04~3abeff8 1001
       1001 http://ppa.launchpad.net/system76/pop/ubuntu hirsute/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Issue/Bug Description:**
Whenever I reboot my computer, I am unable to connect to my network via ethernet. After running `journalctl -xe` I found the culprit to be the r8169 driver.

**Steps to reproduce (if you know):**
- Reboot computer
- Try to connect to network

**Expected behavior:**
- A kernel message will appear stating the following:
`r8169 0000:04:00.0 enp4s0: rtl_rxtx_empty_cond == 0 (loop: 42, delay: 100)`

**Other Notes:**
- I uploaded the system log after running `modprobe r8169` [here](https://pastebin.com/kPA1wM82).
- The only way to solve this that I have discovered is to:
    - Reboot my computer into my Windows 10 install and then reboot back into Pop!_OS, or
    - Install `r8169-dkms` and run `rmmod r8169 && modprobe r8168` after installation
- I have a Gigabyte B450M DS3H with a Ryzen 2600
- Here is the output of `lspci`:
```
04:00.0 Ethernet controller: Realtek Semiconductor Co., Ltd. RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller (rev 0c)
```
- Here is the output of `ip link show`:
```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: enp4s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP mode DEFAULT group default qlen 1000
    link/ether 18:c0:4d:2f:b0:7d brd ff:ff:ff:ff:ff:ff
```

**Comment by jacobgkau** [comment]
This is typically something your hardware manufacturer would need to fix. Does the issue also occur in Ubuntu, or is it specific to Pop!_OS?

---

System76 customers can reach out to [support](https://support.system76.com/) for technical assistance. For non-System76 hardware, you can seek community support on [Reddit](https://reddit.com/r/pop_os/) or [Mattermost](https://chat.pop-os.org).

**Comment by getchoo** [comment]
Yes, it happens in Ubuntu as well as a few other distributions; this is just the first time i’ve gotten annoyed enough to save the log (maybe not the best choice but yeah). I was thinking of trying to install one of the newer kernels to see if maybe there was an update that hasn’t been backported because a friend of mine with the same motherboard said he didn’t run any issues running the 5.13 kernel series on his system.

I also don’t understand how it would be a manufacturer issue though, as I don’t run into any problem with it on Windows, and the proprietary r8168 doesn’t have any problems either, so to me the problem seems to be from the open source r8169 driver - though I could be wrong, i’m just a bit confused.
